ext/node/ops.rs

```diff
@@ -470,7 +470,13 @@ where
   {
     modules_path
   } else {
-    path_resolve(vec![modules_path, name])
+    let orignal = modules_path.clone();
+    let mod_dir = path_resolve(vec![modules_path, name]);
+    if Path::new(&mod_dir).is_dir() {
```

Comment:
Could you please replace `is_dir()` call with `Env::Fs::metadata()`? We don't want to hardcode access to file-system here, because systems like Deno Deploy, will inject another implementation of file system (via the `Env::Fs` type)

Reply:
We should also add `is_dir()` and related methods to https://github.com/denoland/deno/blob/main/ext/node/clippy.toml

Reply:
Might just be me being new to Rust. I don't have a good feel yet for which methods are preferred and which aren't 👀

Reply:
I think we should add the following (all the methods on std::path::Path that interact with the file system https://doc.rust-lang.org/std/path/struct.Path.html):

```toml
{ path = "std::path::Path::canonicalize", reason = "File system operations should be done using NodeFs trait" },
{ path = "std::path::Path::is_dir", reason = "File system operations should be done using NodeFs trait" },
{ path = "std::path::Path::is_file", reason = "File system operations should be done using NodeFs trait" },
{ path = "std::path::Path::is_symlink", reason = "File system operations should be done using NodeFs trait" },
{ path = "std::path::Path::metadata", reason = "File system operations should be done using NodeFs trait" },
{ path = "std::path::Path::read_dir", reason = "File system operations should be done using NodeFs trait" },
{ path = "std::path::Path::read_link", reason = "File system operations should be done using NodeFs trait" },
{ path = "std::path::Path::symlink_metadata", reason = "File system operations should be done using NodeFs trait" },
{ path = "std::path::Path::try_exists", reason = "File system operations should be done using NodeFs trait" },
```

Reply:
My bad here, I should have added these methods to `ext/node/clippy.toml` in a previous PR

Reply:
Updated in #18604 and fixed it in this PR.